pkgs/build-support/docker/layer-strategies/popularity_weighted_top.py

```diff
@@ -0,0 +1,15 @@
+import json
+import sys
+
+available_layers = int(sys.argv[1])
+pkgs = [x.strip() for x in sys.stdin.readlines()]
+
+# Layers [2, limit] have 1 pkg each, of the user's top-most pkgs,
+# and layer 1 has the remainder.
+user_pkgs = pkgs[-(available_layers - 1):]
+layers = [pkgs[:-(available_layers - 1)]]
```

Comment:
Just realized this produces an empty layer when there are less `pkgs` than `available_layers`: `layers = [ [] ]`.

You could check if you have any base layers before initializing `layers`. Here's what I tested locally:
```python
base_pkgs = pkgs[:-(available_layers - 1)]

layers = [base_pkgs] if base_pkgs else []
layers.extend([pkg] for pkg in user_pkgs)
```

Reply:
We should have a regression test for this.